deploy/early-access/reference/apps.md

```diff
@@ -97,3 +97,67 @@ in the repository dropdown. This will redirect you to GitHub to authorize the
 Deno Deploy GitHub app with the selected GitHub account or organization. After
 authorization, you will be redirected back to the app settings page, where you
 can select the newly authorized GitHub repository.
+
+### GitHub events integration
+
+Whenever Deno Deploy<sup>EA</sup> builds an app from a GitHub repository, it
+will send a
+[`repository_dispatch`](https://docs.github.com/en/actions/reference/workflows-and-actions/events-that-trigger-workflows#repository_dispatch)
+event to the repository at the start and end of the build. This allows you to
+trigger GitHub Actions workflows based on the build status.
+
+Deno Deploy will send the following events:
+
+| Event Name                    | Description                                                                |
+| ----------------------------- | -------------------------------------------------------------------------- |
+| `deno_deploy.build.enqueued`  | Sent when a build is enqueued, i.e. when a push is made to the repository. |
+| `deno_deploy.build.cancelled` | Sent when a build is cancelled, either manually or due to a timeout.       |
+| `deno_deploy.build.failed`    | Sent when a build fails.                                                   |
+| `deno_deploy.build.routed`    | Sent when a build completes successfully, and traffic is routed to it.     |
+
+The payload of the event follows the following TypeScript type definition:
+
+```ts
+interface DenoDeployBuildEventPayload {
+  app: {
+    /** The UUID of the Deno Deploy app. */
+    id: string;
+    /** The slug (name) of the Deno Deploy app. */
+    slug: string;
+  };
+  organization: {
+    /** The UUID of the Deno Deploy organization containing the app. */
+    id: string;
+    /** The slug (name) of the Deno Deploy organization containing the app. */
+    slug: string;
+  };
+  revision: {
+    /** The ID of the revision being built. */
+    id: string;
+    /** A URL to view the revision and build status in the Deno Deploy dashboard. */
+    html_url: string;
+    /** The Git commit SHA being built. */
+    git: { sha: string };
+    /** The preview URL the revision is available at, if the build succeeded. */
+    preview_url: string | null;
```

Comment:
This is the revision url right? that's great! Will I get this with `deno_deploy.build.routed`? Also why can this be null? when deployment fails?

Reply:
This will be the preview URL, yes. It will always be set if the event is `deno_deploy.build.routed`. It will be `null` for all other events.